fhir-tools/src/main/java/com/ibm/fhir/tools/LanguageRegistryGenerator.java

```diff
@@ -0,0 +1,324 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.tools;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class that generates the contents of the static initializer methods of the LanguageRegistry class
+ * from the contents of the language-subtag-registry.txt file.
+ * It does not generate the entire Java class, but rather just the sets/maps that need to be built by
+ * the static initializer methods. The generated code can be copied/pasted into those methods of the
+ * LanguageRegistry class.
+ */
+public class LanguageRegistryGenerator {
+
+    private static final String FILE_DATE_ROW = "File-Date:";
+    private static final String RECORD_DELIMITER = "%%";
+    private static final String ROW_DELIMITER = ": ";
+    private static final String TYPE = "Type";
+    private static final String LANGUAGE = "language";
+    private static final String EXTLANG = "extlang";
+    private static final String SCRIPT = "script";
+    private static final String REGION = "region";
+    private static final String VARIANT = "variant";
+    private static final String GRANDFATHERED = "grandfathered";
+    private static final String DEPRECATED = "Deprecated";
+    private static final String SUBTAG = "Subtag";
+    private static final String PREFIX = "Prefix";
+    private static final String TAG = "Tag";
+    private static final String PRIVATE_USE_LANG_SUBTAG = "qaa..qtz";
+    private static final String PRIVATE_USE_SCRIPT_SUBTAG = "Qaaa..Qabx";
+    private static final String PRIVATE_USE_REGION_SUBTAG_1 = "QM..QZ";
+    private static final String PRIVATE_USE_REGION_SUBTAG_2 = "XA..XZ";
+    private final List<Map<String, String>> languageSubTagRegistry;
+
+    /**
+     * Instantiates a LanguageTagGenerator.
+     * 
+     * @param languageSubTagRegistry
+     */
+    public LanguageRegistryGenerator(List<Map<String, String>> languageSubTagRegistry) {
+        this.languageSubTagRegistry = languageSubTagRegistry;
+    }
+
+    /**
+     * Generates the Language Tag classes.
+     */
```

Comment:
Add @param tags (same comment for other methods in this class as well)

Reply:
Fixed.